app/policies/namespace_policy.rb

```diff
@@ -42,10 +42,15 @@ def index?
 
   def create?
     raise Pundit::NotAuthorizedError, "must be logged in" unless user
-    (APP_CONFIG.enabled?("user_permission.manage_namespace") || user.admin?) && push?
+    (APP_CONFIG.enabled?("user_permission.create_namespace") || user.admin?) && push?
+  end
+
+  def update?
+    raise Pundit::NotAuthorizedError, "must be logged in" unless user
+    (user.admin? || (APP_CONFIG.enabled?("user_permission.manage_namespace") &&
+                     namespace.team.owners.exists?(user.id))) && push?
```

Comment:
Sorry, my bad. I got confused with the `owner?` functions in `namespace_helper.rb` and `team_policy.rb`. The code is perfectly fine.